ruma-common/src/push.rs

```diff
@@ -206,6 +211,159 @@ impl Serialize for Action {
     }
 }
 
+/// Optional prefix used by `RoomMemberCountIs`
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum RoomMemberCountPrefix {
+    /// No prefix
+    None,
+    /// Equals
+    Eq,
```

Comment:
In general we don't store semantically irrelevant details like whether a field is absent, null or has its default value set. We should apply the same here and have only `Eq`, not `None`.

Reply:
I thought it could be relevant because they're serialized differently.

For example, suppose there was a file that included data represented by `PushCondition`, and a GUI for editing that file. The user could reasonably expect the app to only change parts of the file that were changed in the GUI. But the app would be unable to remember which exact text representation was used before, and would have to normalize everything.

This is up to you, of course: I don't know much about the way these types are intended to be used yet, so [I have no strong feelings one way or the other](https://www.youtube.com/watch?v=ussCHoQttyQ) 🙂

Reply:
Supporting that kind of use case is just way too much work. For example, simply using serde's builtin `Deserialize` & `Serialize` implementations for `Option` already breaks input -> deserialize -> serialize -> same output, because it represents both missing fields and fields with a `null` value as `None`. Furthermore, unknown keys are ignored by default and not serialized again. That is why we don't try to uniquely represent every possible serialized form in our custom Rust types either, only semantically relevant bits.

Reply:
Makes sense!